### PR TITLE
HSEARCH-4791 + HSEARCH-4783 follow-up - upgrade to ORM 6.2.0.CR2 for -orm6 artifacts and fixes for tests on non-H2 databases for orm6/jakarta artifacts

### PR DESCRIPTION
**Pull request by yrodiere**
Follows up on #3377 

https://hibernate.atlassian.net/browse/HSEARCH-4791
https://hibernate.atlassian.net/browse/HSEARCH-4783

Draft, because I wasn't able to reproduce the Oracle failures locally, even though they happen pretty much all the time on CI. Let's run this on CI and see if maybe some of my changes fixed the Oracle test failures inadvertently...

### CODE REVIEW

File: integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/association/bytype/onetoone/ownedbycontained/AutomaticIndexingOneToOneOwnedByContainedLazyOnContainedSideIT.java [hunk 1]
```diff
@@ -89,6 +90,11 @@ protected boolean isAssociationLazyOnContainingSide() {
 	public void setup(OrmSetupHelper.SetupContext setupContext,
 			ReusableOrmSetupHolder.DataClearConfig dataClearConfig) {
 		super.setup( setupContext, dataClearConfig );
+		// Avoid problems with deep chains of eager associations in ORM 6
+		// See https://github.com/hibernate/hibernate-orm/blob/main/migration-guide.adoc#fetch-behaviour-change
```
Comment (marko-bekhta) [inline integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/association/bytype/onetoone/ownedbycontained/AutomaticIndexingOneToOneOwnedByContainedLazyOnContainedSideIT.java:94]: I wanted to learn more about the changes in fetching, and the link looks broken now 😢.
The somewhat related to the fetching section in the 6.0 guide - https://github.com/hibernate/hibernate-orm/blob/6.0/migration-guide.adoc#fetch-circularity-determination is a bit confusing (unfinished or repeating sentences) and sadly I'm not with enough knowledge to update it 😞

Comment (yrodiere) [reply integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/association/bytype/onetoone/ownedbycontained/AutomaticIndexingOneToOneOwnedByContainedLazyOnContainedSideIT.java:94]: Yes I guess that's the right section, I'll update the links. As to updating the migration guide, maybe ping Christian about that?

Comment (marko-bekhta) [reply integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/association/bytype/onetoone/ownedbycontained/AutomaticIndexingOneToOneOwnedByContainedLazyOnContainedSideIT.java:94]: thanks, will do 😃

Comment (yrodiere) [reply integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/association/bytype/onetoone/ownedbycontained/AutomaticIndexingOneToOneOwnedByContainedLazyOnContainedSideIT.java:94]: Anyway the gist of it is "if you have multiple circular eager associations, ORM will now (correctly) follow them and your queries will become exponentially larger as the depth increases, so you'd better limit the depth"

**Comment by yrodiere** [comment]
Note the MSSQL failure in https://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/PR-3396/16/tests is caused by a flaky test; see https://hibernate.atlassian.net/browse/HSEARCH-4774.

Hoping for all tests to pass with CockroachDB, then we can merge. :crossed_fingers:

**Comment by marko-bekhta** [comment]
thanks! merging 😃

**Pull request merged**